src/main/java/com/julienviet/pgclient/Numeric.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2017 Julien Viet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.julienviet.pgclient;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Objects;
+
+/**
+ * The Postgres <i>NUMERIC</i> type.
+ */
+public final class Numeric extends Number {
+
+  /**
+   * Constant for the {@code NaN} value.
+   */
+  public static final Numeric NaN = new Numeric(null);
```

Comment:
You also might consider to handle `Infinity` and `-Infinity`.

Reply:
does Postgres support them ? according to the documentation, only _Nan_ makes sense

Reply:
NaN for Numeric only

but the floating-point types have several special values:

Infinity
-Infinity
NaN

Reply:
are we talking of PG or Java types ?

Reply:
PG types

Reply:
https://www.postgresql.org/docs/9.6/static/datatype-numeric.html

Reply:
ah cool, it was not clear from the docs I read, will update it

Reply:
so actually infinite values are not supported for numeric (https://stackoverflow.com/questions/45603040/does-postgresql-numeric-type-support-infinity-and-infinity), so I don't think that we should handle them in this class except than rejecting the invalid values when creating a numeric from a Number